### PR TITLE
options: watch-later-options changes

### PR DESCRIPTION
**Pull request by kasper93**
--border is there, so for consistency --title-bar should be too.

### CODE REVIEW

**Comment by Dudemanguy** [comment]
Should be mentioned in `interface-changes`.

**Comment by kasper93** [comment]
sure, added.

**Comment by christoph-heinrich** [comment]
`fullscreen` and `pause` could also be removed.

**Comment by low-batt** [comment]
Seems inconsistent that `sub-visibility` is included, but `secondary-sub-visibility` is not.

I do not understand the requirement cited for removing `ontop`, namely that saved options be "content related". To me "Watch Later" is about resuming viewing where you left off. If you had the window set to be on top, why would you want that setting to be dropped? And as already pointed out, if this is the requirement then `fullscreen` and `pause` should be removed as well.

If `fullscreen` is removed then the [RESUMING PLAYBACK](https://mpv.io/manual/stable/#resuming-playback) section of the manual will need to be updated as it mentions restoring the fullscreen state.

**Comment by kasper93** [comment]
I initially only wanted to fix inconsistency between `title-bar` and `border` which are tightly connected options and one is saved and the other not. But as always with such changes a lot voices suggested another options and we are again at the point of no return.

> I do not understand the requirement cited for removing ontop, namely that saved options be "content related". To me "Watch Later" is about resuming viewing where you left off. If you had the window set to be on top, why would you want that setting to be dropped? And as already pointed out, if this is the requirement then fullscreen and pause should be removed as well.

The question is if you watched something few days ago, while doing something else and enabled ontop, is this really relevant now to restore it? It is global player state and indeed same as fullscreen should be set in config. We want to "resume playback" and not "restore player state".

The most basic would be `watch-later-options=start`, but of course it makes sense to restore also track selection, image correction and few other video/audio specific parameters. The option is even called `save-position-on-quit`, but it does lot's more.

In short resuming playback in my opining should start playback where you left of, but not necessarily restore player config as this is not really related to the content you are resuming playing.

There is of course a line and depends where you want to put it. I for one like `pause`, because it makes easy to restart player to reload config and start at the same frame, but I guess I used to it.

**Comment by kasper93** [comment]
Rebased with added suggestions. I don't have strong opinion on this topic, so will let you decide what to do with this. I only wanted to fix inconsistency with --border. But other changes makes sense too, so we can have them if all agree.

**Comment by guidocella** [comment]
I don't think it makes sense to save the `video-margin-` options because they are meant for the OSC, with `--script-opts=osc-boxvideo=yes,osc-visibility=always` `video-margin-ratio-bottom` gets saved and it is restored the nextime even with different/default OSC options, which is annoying.

**Comment by kasper93** [comment]
> I don't think it makes sense to save the `video-margin-` options because they are meant for the OSC, with `--script-opts=osc-boxvideo=yes,osc-visibility=always` `video-margin-ratio-bottom` gets saved and it is restored the nextime even with different/default OSC options, which is annoying.

Removed.

**Comment by hooke007** [comment]
Is there any use case of `osd-level` for single file?

**Comment by kasper93** [comment]
Removed `osd-level` too.

**Review by Dudemanguy — APPROVED** [review]
This got lost, but I think it's a strict improvement so might as well pull this in now after a rebase.

**Comment by low-batt** [comment]
In [RESUMING PLAYBACK](https://mpv.io/manual/stable/#resuming-playback) the manual says:
> mpv also stores options other than the playback position when they have been modified after playback began, for example the volume and the fullscreen state

As this commit removes `fullscreen` from the `watch-later-options` default, that example should be updated.

In [watch-later-options](https://mpv.io/manual/master/#options-watch-later-options) the manual shows this example:
> - --watch-later-options-remove=fullscreen The fullscreen state won't be saved to watch later files.

That example should also be updated.

**Comment by hooke007** [comment]
Ideally, add all the props it support into the manual. Users don't know which prop can & cannot be used.

**Comment by Jules-A** [comment]
What about creating a new command that specifically tries to recreate the player's state? I'm not really a fan of removing such options unless there's a solution in place to restore that functionality...

**Comment by christoph-heinrich** [comment]
> I'm not really a fan of removing such options unless there's a solution in place to restore that functionality...

e.g. `--watch-later-options-append=fullscreen`

**Comment by Jules-A** [comment]
> e.g. `--watch-later-options-append=fullscreen`

Sure but it gets really messy when you have to re-add like 5+ items. Also it's not really easy for people new to MPV. Another thing is what do you do if you want to restore playback state but not the playback position?

**Comment by hooke007** [comment]
> Another thing is what do you do if you want to restore playback state but not the playback position?

remove `start`

**Comment by Jules-A** [comment]
> remove `start`

You mean you still use save-position-on-quit but remove start? That's rather hilarious

**Comment by hooke007** [comment]
> Another thing is what do you do if you want to restore playback state but not the playback position?

I'm confused. Isn't the 'hilarious' question asked by you?

**Comment by guidocella** [comment]
> Ideally, add all the props it support into the manual. Users don't know which prop can & cannot be used.

You can see it with `mpv --help=watch-later-options`, though it's not very discoverable.

> In [watch-later-options](https://mpv.io/manual/master/#options-watch-later-options) the manual shows this example:
>
> > * --watch-later-options-remove=fullscreen The fullscreen state won't be saved to watch later files.
>
> That example should also be updated.

If you update that kasper you might as well replace `--watch-later-options-clr` with `--watch-later-options=start` since I wrote the `clr` example when `start` was implicitly included and removing `start` is undesirable for most users.

**Comment by Jules-A** [comment]
> I'm confused. Isn't the 'hilarious' question asked by you?

The option is called save-position-on-quit even if you specifically don't want to save position, which is what I find funny.

**Comment by guidocella** [comment]
> The option is called save-position-on-quit even if you specifically don't want to save position, which is what I find funny.

`--save-position-on-quit` existed long before `--watch-later-options`.

**Comment by hooke007** [comment]
You are not the first one point out the option's name is unreasonable.
When I am the newbie, I was annoyed why it restored so many irrelevant props.(I wanted position only)

**Comment by kasper93** [comment]
For some reason, I was unsubscribed from this thread, didn't see your comments. Will try to address them now.

> When I was the newbie, I was annoyed why it restored so many irrelevant props.(I wanted position only)

We all did... It is one of those options, when you find it you just keep `--watch-later-options=start` in your config to not be annoyed, but some weird option restores.

File: DOCS/interface-changes.rst [hunk 1]
```diff
@@ -115,6 +115,9 @@ Interface changes
     - add hdr metadata to `video-params` property
     - remove `hdr-metadata` property
     - add `--target-gamut`
+    - add `video-params-dyn`, `video-dec-params-dyn`, `video-out-params-dyn`
```
Comment (Dudemanguy) [inline DOCS/interface-changes.rst:118]: Crept in from the other branch.

Comment (kasper93) [reply DOCS/interface-changes.rst]: oops, fixed.

**Review by Dudemanguy — APPROVED** [review]
LGTM. Will merge later if nobody has any further comments.

**Comment by mlindner** [comment]
This is a bad change given right now it's cumbersome to add these options back in. You have to use --watch-later-options-add and you get spammed with warnings about the option being deprecated.

**Comment by christoph-heinrich** [comment]
> You have to use --watch-later-options-add and you get spammed with warnings about the option being deprecated.

You could use `watch-later-options-append` instead.
